h3/src/config.rs

```diff
@@ -0,0 +1,43 @@
+use crate::proto::varint::VarInt;
+
+/// Configures the HTTP/3 connection
+#[derive(Debug, Clone, Copy)]
+pub struct Config {
```

Comment:
Is this type meant to be publicly exposed? If so, having all it's fields public will make it impossible to add any new options without it being a breaking change. This is why the builder pattern is used the way it is in `h2` and `hyper`.

Reply:
I've marked the struct as non-excaustive for now.

It isn't really public either, as it is never exported `mod config`.

This is a pattern to allow using a type as a return value of a public function, without getting the "can not leak private type" While it is public, the module/symbol is never exported so the user can't mention it or store it in a struct, only use it with local variables given from a return value.

Reply:
I've made the fields public, and added getters for those need by webtransport